lib/wasix/src/runtime/task_manager/waker.rs

```diff
@@ -0,0 +1,70 @@
+use std::{
+    sync::{mpsc, Arc},
+    task::{Context, Poll, RawWaker, RawWakerVTable, Waker},
+};
+
+use futures::Future;
+
+pub struct InlineWaker {
+    waker_rx: mpsc::Receiver<()>,
+    waker_tx: mpsc::Sender<()>,
```

Comment:
Can we split this into two types?
Primarily: can the actual waker just hold the sender, so that dropping it will also make the receiver fail?